File: js/programmers/level.1/prgm_138477_1.js
function solution(k, score) {
  let honor = [];
  let answer = [];

  score.forEach((num) => {
    honor.push(num);
    honor.sort((a, b) => b - a);

    if (honor.length >= k) {
      answer.push(honor[k - 1]);
    } else {
      answer.push(honor.at(-1));
    }
  });
  return answer;
}

File: python/programmers/level.1/prgm_138477_1.py
def solution(k, score):
    hall_of_fame = []
    result = []

    for i in score:
        hall_of_fame.append(i)
        hall_of_fame = sorted(hall_of_fame)[::-1][:k]
        result.append(hall_of_fame[-1])

    return result
